src/Api/Test/ClientTest.php

```diff
@@ -0,0 +1,572 @@
+            '@type' => 'hydra:Error',
+            'hydra:title' => 'An error occurred',
+            'hydra:description' => 'Script "1" does not exist',
+        ]);
+    }
+
+    public function testDeleteClient(): void
+    {
+        $httpClient = $this->authenticate();
+        $timestamp = $this->getTimestamp();
+        $httpClient->request('POST', '/api/clients', [
+            'json' => [
+                'isActive' => true,
+                'maxParallelJobs' => 1,
+                'name' => 'client' . $timestamp,
+                'owner' => 1,
+                'quota' => - 1
+            ]
+        ]);
+        $iri = $this->findIriBy(Client::class, [
+            'name' => 'client' . $timestamp
+        ]);
+        $response = $httpClient->request('DELETE', $iri);
+        $this->assertResponseIsSuccessful();
+    }
+
+    public function testDeleteClientNotFound(): void
+    {
+        $httpClient = $this->authenticate();
+        $response = $httpClient->request('DELETE', 'api/clients/0');
+        $this->assertResponseStatusCodeSame(404);
+        $this->assertResponseHeaderSame('content-type', 'application/ld+json; charset=utf-8');
+        $this->assertJsonContains([
+            '@context' => '/api/contexts/Error',
+            '@type' => 'hydra:Error',
+            'hydra:title' => 'An error occurred',
+            'hydra:description' => 'The client "0" does not exist.',
+        ]);
+    }
+
+    public function testGetCollection(): void
+    {
+        $httpClient = $this->authenticate();
+        $timestamp = $this->getTimestamp();
+        $httpClient->request('POST', '/api/clients', [
+            'json' => [
+                'isActive' => true,
+                'maxParallelJobs' => 1,
+                'name' => 'client' . $timestamp,
+                'owner' => 1,
+                'quota' => - 1
+            ]
+        ]);
+        $response = $httpClient->request('GET', '/api/clients');
+        $this->assertResponseIsSuccessful();
+        $this->assertResponseHeaderSame('content-type', 'application/ld+json; charset=utf-8');
+        $this->assertJsonContains([
+        '@context' => '/api/contexts/Client',
+        '@type' => 'hydra:Collection',
+        '@id' => '/api/clients',
+        ]);
+    }
+
+    public function testGetCollectionUnauthenticated(): void
+    {
+        $response = static::createClient()->request('GET', '/api/clients');
+        $this->assertResponseStatusCodeSame(401);
+    }
+
+    public function testGetInvalidItem(): void
+    {
+        $httpClient = $this->authenticate();
+        $timestamp = $this->getTimestamp();
+        $httpClient->request('POST', '/api/clients', [
+            'json' => [
+                'isActive' => true,
+                'maxParallelJobs' => 1,
+                'name' => 'client' . $timestamp,
+                'owner' => 1,
+                'quota' => - 1
+            ]
+        ]);
+        $response = $httpClient->request('GET', '/api/clients/' . $timestamp);
```

Comment:
Why do you use $timestamp as client id?

If you want to referene a non existing client just use a fixed high number or something like that. I know the timestamp is in fact a hi number but semantic also counts. Anyone seeing this code will think that someone the timestamp has something to do with what you are doing and is not. Or am I wrong?